### PR TITLE
KIK-8246-delete-account

### PR DESCRIPTION
**Pull request by shayb**
change layout
and call api
with alert warning

### CODE REVIEW

File: sample/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -15,7 +15,11 @@
     <string name="transaction">Transaction</string>
     <string name="send_transaction">Send Transaction</string>
     <string name="get_kin">Get KIN</string>
-    <string name="create_account">Create Account</string>
+    <string name="delete_wallet">Delete Wallet</string>
+    <string name="delete_wallet_warning">Are you sure you want to delete your wallet?\n\nDelete without backed up will cause permanent funds lost</string>
```
Comment (berryve) [inline sample/src/main/res/values/strings.xml]: Are you sure you want to delete your wallet?\n\nWarning: deleting a wallet when it hasn't been backed up will cause funds to be lost. Use Export Key Store to backup your wallet.

File: sample/src/main/java/kin/sdk/core/sample/ViewUtils.java [hunk 2]
```diff
@@ -12,12 +12,17 @@
 
 public class ViewUtils {
 
+    public interface onAlertClicked {
```
Comment (berryve) [inline sample/src/main/java/kin/sdk/core/sample/ViewUtils.java]: OnConfirmListener

File: sample/src/main/java/kin/sdk/core/sample/ViewUtils.java [hunk 3]
```diff
@@ -28,6 +33,32 @@ public static void alert(Context context, String message) {
         dialog.show();
     }
 
+    public static void confirmAlert(Context context, String message, String button, onAlertClicked onAlertClicked) {
```
Comment (berryve) [inline sample/src/main/java/kin/sdk/core/sample/ViewUtils.java]: 1. method name can be `alert`.
2. rename `String button` to `String confirmationMessage`
3. `OnConfirmListener confirmationListener`

File: sample/src/main/res/values/strings.xml (hunk 1 above)
Comment (shayb) [reply sample/src/main/res/values/strings.xml]: this is too long for this dialog
need to give the shortest message that will give this info.
no word of export in ios - no need to add it here impo

File: sample/src/main/java/kin/sdk/core/sample/ViewUtils.java (hunk 3 above)
Comment (shayb) [reply sample/src/main/java/kin/sdk/core/sample/ViewUtils.java]: 👍

File: sample/src/main/java/kin/sdk/core/sample/ViewUtils.java [hunk 4]
```diff
@@ -28,6 +33,32 @@ public static void alert(Context context, String message) {
...
+        builder = new AlertDialog.Builder(context);
+        if (TextUtils.isEmpty(message)) {
+            message = context.getResources().getString(R.string.error_no_message);
+        }
+        builder.setView(buildAlertView(context, message));
+        dialog = builder.create();
+        dialog.setCancelable(true);
+        if (TextUtils.isEmpty(button)) {
+            button = context.getResources().getString(R.string.ok);
+        }
```
Comment (berryve) [inline sample/src/main/java/kin/sdk/core/sample/ViewUtils.java]: code duplication with original alert method

File: sample/src/main/java/kin/sdk/core/sample/ViewUtils.java (hunk 3 above)
Comment (OrenZak) [inline sample/src/main/java/kin/sdk/core/sample/ViewUtils.java]: A suggestion: create BaseDialog that extends from Dialog and receive in the constructor title, message, confirmListener.
and create DeleteDialog that extends BaseDialog.

File: sample/src/main/res/values/strings.xml (hunk 1 above)
Comment (OrenZak) [reply sample/src/main/res/values/strings.xml]: agree with @shayb

File: sample/src/main/java/kin/sdk/core/sample/CreateAccountActivity.java [hunk 5]
```diff
@@ -51,7 +51,7 @@ Intent getBackIntent() {
 
     @Override
     int getActionBarTitleRes() {
-        return R.string.create_account;
+        return R.string.create_wallet;
```
Comment (OrenZak) [inline sample/src/main/java/kin/sdk/core/sample/CreateAccountActivity.java]: So maybe we can rename the activity name to CreateWalletActivity?

Comment (shayb) [reply sample/src/main/java/kin/sdk/core/sample/CreateAccountActivity.java]: 👍

File: sample/src/main/java/kin/sdk/core/sample/ExportKeystoreActivity.java [hunk 6]
```diff
@@ -43,7 +45,7 @@ private void initWidgets() {
 
         findViewById(R.id.copy_btn).setOnClickListener(view -> {
             selectAll();
-            ViewUtils.copyToClipboard(this, outputTextView.getText());
+            BaseDialog.copyToClipboard(this, outputTextView.getText());
```
Comment (berryve) [inline sample/src/main/java/kin/sdk/core/sample/ExportKeystoreActivity.java]: copyToClipboard has nothing to do with the BaseDialog.
I think we should keep ViewUtils just for that method

File: sample/src/main/java/kin/sdk/core/sample/kin/sdk/core/sample/dialog/BaseDialog.java [hunk 7]
```diff
@@ -9,26 +9,55 @@
...
         dialog.setCancelable(true);
-        dialog.setButton(DialogInterface.BUTTON_POSITIVE, context.getResources().getString(R.string.ok),
+    }
+
+    protected void setMessage(String message) {
+        dialog.setView(buildMessageView(message));
+    }
+
+    public void show() {
+        if (!((Activity) context).isFinishing()) {
```
Comment (OrenZak) [inline sample/src/main/java/kin/sdk/core/sample/kin/sdk/core/sample/dialog/BaseDialog.java]: maybe it's good to check context != null before

File: sample/src/main/java/kin/sdk/core/sample/kin/sdk/core/sample/dialog/ConfirmDialog.java [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+
+    public interface OnConfirmListener {
+
+        void onConfirm();
+    }
+
+    private OnConfirmListener confirmListener;
+
+    public ConfirmDialog(Context context, String message, String confirmationText,
+        OnConfirmListener confirmListener) {
```
Comment (OrenZak) [inline sample/src/main/java/kin/sdk/core/sample/kin/sdk/core/sample/dialog/ConfirmDialog.java]: maybe we can create a setConfirmationText(String text) and remove String confirmationText  from the constructor.
with a default test "OK"/"Confirm"

File: sample/src/main/java/kin/sdk/core/sample/kin/sdk/core/sample/dialog/KinAlertDialog.java [hunk 9]
```diff
@@ -0,0 +1,108 @@
...
+    public void show() {
+        if (context != null && !((Activity) context).isFinishing()) {
+            dialog.show();
+        }
+    }
+
+    private void setPositiveButtonText(String text){
+        if(TextUtils.isEmpty(text)) {
+            positiveButtonText = context.getResources().getString(R.string.ok);
+        }else {
```
Comment (berryve) [inline sample/src/main/java/kin/sdk/core/sample/kin/sdk/core/sample/dialog/KinAlertDialog.java]: format

**Pull request merged**